libconnman-qt/marshalutils.cpp

```diff
@@ -31,11 +31,16 @@
  */
 
 #include <QDebug>
-#include <QDBusMetaType>
+#include <QDBusArgument> 
 #include "vpnconnection.h"
 
 #include "marshalutils.h"
 
+#if defined(__clang__) &&  __clang_major__ >= 11 || __GNUC__ >= 12
+Q_DBUS_EXPORT const QDBusArgument &operator>>(const QDBusArgument &a, RouteStructure &v);
+Q_DBUS_EXPORT QDBusArgument &operator<<(QDBusArgument &a, const RouteStructure &v);
```

Comment:
This looks strange to me in may ways.  First the Q_DBUS_EXPORT not being a documented, looks like it should end up as Q_DECL_IMPORT/EXPORT. Then again the symbol visibilities shouldn't generally depend on compiler. And then again generally the visibilities should only matter on something using this from a shared library. And not sure does the library here get even compiled with hidden symbols by default. There are no Q_DECL_... usage elsewhere so I'm guessing no. And finally this is exporting something from the anon namespace which by definition should be internal to the file.

I'm having a feeling this might be fixing something by accident.

Reply:
> Perhaps, but even without that this would be having different declaration for content inside the anon namespace.
> 
> There was a small compiler error snippet in the commit message. Would you have the bigger error message available in case it helps figuring out what's the problem.

Here's a full log: http://errors.yoctoproject.org/Errors/Details/655056/

Reply:
@pvuorela And another one: https://pastebin.ubuntu.com/p/D5RRy77K8v/

Reply:
@pvuorela Any thoughts based on above logs? I got a build system with GCC 12 here to test if needed.

Reply:
Thanks. Tested also local build with qt6 and the current version here didn't work.

But tried a minor adjustment which could make more sense. Suggest:
- remove this export thing
- move the empty namespace declaration two methods downwards so the dbus argument operators are not inside it.

Reply:
> Thanks. Tested also local build with qt6 and the current version here didn't work.
> 
> But tried a minor adjustment which could make more sense. Suggest:
> 
> * remove this export thing
> * move the empty namespace declaration two methods downwards so the dbus argument operators are not inside it.

Sorry I'm not that great when it comes to C(++) ;)
You mean like this? https://github.com/webOS-ports/libconnman-qt/commit/eb38903048e90d6d7b9c323ad131ff8c3bbd7c31

Reply:
Same effect, yes. The pre-declarations are unnecessary there, though, when the implementations come right after them.